src/lib/dataEntities/transactionRequest.ts

```diff
@@ -0,0 +1,84 @@
+import { TransactionRequest } from '@ethersproject/providers'
+import { BigNumber } from 'ethers'
+
+/**
+ * A transaction request for a transaction that will trigger some sort of
+ * execution on the L2
+ */
+export interface L1ToL2TransactionRequest {
+  /**
+   * Core fields needed to forma transaction request: `to`, `data`, `from`
+   */
+  txRequestCore: Required<Pick<TransactionRequest, 'to' | 'data' | 'value'>>
```

Comment:
could argue that the gas field also defines the behaviour, depends if you assume this request is pre or post estimation.
that said, instead of having gas here it seems like a fair assumption that its whatever amount of gas needed for the tx not to revert
```suggestion
   * Core fields needed to form a transaction request: `from`, `to`, `data`, `value`, 'chainId', 'nonce'
   */
  txRequestCore: Required<Pick<TransactionRequest, 'from' | 'to' | 'data' | 'value' | 'chainId' | 'nonce'>>
```